src/base/moduleresolver.lua

```diff
@@ -0,0 +1,90 @@
+--
+-- moduleresolver.lua
+-- 
+-- Provides a strategy for module resolution to modules that are placed within the default module's directory. 
+-- 
+-- Copyright (c) 2002-2014 Jason Perkins and the Premake project
+--
+
+    local p = premake
+
+    p.moduleresolver = {}
+
+    local moduleresolver = p.moduleresolver
+
+---
+-- Check whether the module is either a path to a module or a module's name.
+--
+
+    function moduleresolver.isModulePath(module)
```

Comment:
Maybe we should have a `path.hasdir()` or something more general? I know we do a similar test elsewhere, and there is nothing module-specific about this?